File: Practica1Ruby/pkg/Practica1Ruby-0.0.1/lib/Carrera.rb
public
class Carrera

	public
	def empezar() 
    puts "Comienza la carrera"
    for i in @bicicletas
      i.correr
    end
	end

	def initialize()
		@bicicletas
		@numBicicletas
	end
  
  def clone()
    
  end

end

File: Practica1/src/practica1/Carrera.java
package practica1;
import java.util.*;

public abstract class Carrera {
    
        protected int numBicicletas;
        protected ArrayList<Bicicleta> bicicletas;
        
        Carrera(){
            bicicletas = new ArrayList<>();
        }
        
        public void empezar(){
            
            System.out.println("Comienza la carrera");
            for (int i=0; i<bicicletas.size(); i++){
                bicicletas.get(i).start();
            }
            
            try {
                Thread.sleep(61000);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
                return;
            }
        }

        
}